### PR TITLE
fix(@angular-devkit/build-angular): remove pure_getters

### LINKED ISSUE #9231: 1.6.1+ build optimizer breaking bootstrap carousel transition effects
### Versions
```    _                      _                 ____ _     ___
   / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
  / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
 / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
/_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
               |___/

Angular CLI: 1.6.4
Node: 8.1.4
OS: win32 x64
Angular: 5.2.0
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cli: 1.6.4
@angular-devkit/build-optimizer: 0.0.38
@angular-devkit/core: 0.0.25
@angular-devkit/schematics: 0.0.48
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.4
@schematics/angular: 0.1.13
@schematics/schematics: 0.0.13
typescript: 2.5.3
webpack: 3.10.0
```

### Repro steps

* Step 1 - Download project from https://github.com/rajravat/angular-bug
* Step 2 - Run ng serve --prod 

### Observed behavior
```
carousel transitions are not behaving correctly.
```

### Desired behavior
The transitions for the bootstrap carousel are very janky, i believe this is caused by the build optimizer and has been happening since cli 1.6.1+ (works fine with 1.6.0 and below), if you run ng server --prod --build-optimizer=false then the carousel transitions display correctly.

Is this a bug in the build optimizer? i'm currently working around this issue by downgrading to cli 1.6.0

### Mention any other details that might be useful (optional)
The bootstrap version is 3.1.0

### PR DESCRIPTION
**Pull request by filipesilva**
When we first started using Build Optimizer, we saw a lot of the savings were tied to using the Uglify/Terser `pure_getters` option.

This was intimately related with the structure and shape of the Angular codebase. The measurements we did at the time on angular.io showed a significant size reduction, from 1mb to about 600kb. Of these roughly 150kb were tied to using `pure_getters` if I remember correctly.

Meanwhile the Angular codebase has changed significantly and I don't really see these savings anymore, so I don't think it makes sense to keep it on given that it is known to cause problems with some libraries.

Closes #9231, #11439, #12096, #12128.

### CODE REVIEW

**Comment by filipesilva** [comment]
Blocked on https://github.com/angular/angular/pull/29929.

File: packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts [hunk 1]
```diff
@@ -254,6 +254,20 @@ export function getCommonConfig(wco: WebpackConfigOptions): Configuration {
...
+    let angularGlobalDefinitions = {
+      ngDevMode: false,
+      ngI18nClosureMode: false,
+    };
+
+    try {
+      // Try to load known global definitions from @angular/compiler-cli.
+      // tslint:disable-next-line:no-implicit-dependencies
+      const GLOBAL_DEFS_FOR_TERSER = require('@angular/compiler-cli').GLOBAL_DEFS_FOR_TERSER;
+      angularGlobalDefinitions = GLOBAL_DEFS_FOR_TERSER;
```
Comment (clydin) [inline packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts]: If `GLOBAL_DEFS_FOR_TERSER` doesn't exist, won't this wipe out the defaults?  I think this should probably check for undefined and only set if truthy.

Comment (filipesilva) [reply packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts]: Yes, that's absolutely right. Will change.

File: packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts [hunk 2]
```diff
@@ -254,6 +254,20 @@ export function getCommonConfig(wco: WebpackConfigOptions): Configuration {
...
+    } catch (_error) {
```
[the remark this answers is not in the document]
Comment (filipesilva) [reply packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts]: Yeah I thought so as well, but I was getting squigglies on my editor. Will change and see if the build fails.

**Pull request merged**